medicines/web/src/components/search-filter/index.tsx

```diff
@@ -49,78 +60,103 @@ const AccessibleHeading = styled.h3`
       <div className="checkbox">
         <Checkbox
           id={id}
-          name="doc"
-          value={docTypeForThisCheckbox}
-          checked={currentlyEnabledDocTypes.includes(docTypeForThisCheckbox)}
+          name={query}
+          value={value}
+          checked={checked}
           onChange={toggleDocTypeForThisCheckbox}
         />
       </div>
       <label htmlFor={id}>
-        {name} ({docTypeForThisCheckbox.toUpperCase()})
+        {label} ({value.toUpperCase()})
       </label>
     </div>
   );
 };
 
 const SearchFilter: React.FC<ISearchFilterProps> = (props) => {
-  const [checkedFilters, setCheckedFilters] = React.useState(
+  const [checkedDocFilters, setCheckedDocFilters] = useState(
     props.currentlyEnabledDocTypes,
   );
+  const [checkedTerritoryFilters, setCheckedTerritoryFilters] = useState(
+    props.currentlyEnabledTerritoryTypes,
+  );
   const submitButton = useRef(null);
   const filterHeader = useRef(null);
 
-  const generateCheckboxFor = (docType: DocType, name: string) => (
-    <DocTypeCheckbox
-      toggleDocType={toggleDocType}
-      currentlyEnabledDocTypes={checkedFilters}
-      docTypeForThisCheckbox={docType}
-      name={name}
-    />
-  );
+  const generateDocTypeCheckboxFor = (docType: DocType, name: string) => {
+    const toggleDoc = (docType) =>
+      toggleFilter(docType, checkedDocFilters, setCheckedDocFilters);
+    return (
+      <FilterCheckbox
+        toggleFilter={toggleDoc}
+        value={docType}
+        label={name}
+        query={SearchType.Doc}
+        checked={checkedDocFilters.includes(docType)}
+      />
+    );
+  };
+
+  const generateTerritoryTypeCheckboxFor = (
+    territoryType: TerritoryType,
+    name: string,
+  ) => {
+    const toggleTerritory = (territoryType) =>
+      toggleFilter(
+        territoryType,
+        checkedTerritoryFilters,
+        setCheckedTerritoryFilters,
+      );
+    return (
+      <FilterCheckbox
+        toggleFilter={toggleTerritory}
+        value={territoryType}
+        label={name}
+        query={SearchType.Territory}
+        checked={checkedTerritoryFilters.includes(territoryType)}
+      />
+    );
+  };
 
-  const toggleDocType = (docTypeToToggle) => {
-    const enabledDocTypes = Array.from(checkedFilters);
-    if (enabledDocTypes.includes(docTypeToToggle)) {
-      const docTypeIndex = enabledDocTypes.indexOf(docTypeToToggle);
-      enabledDocTypes.splice(docTypeIndex, 1);
+  const toggleFilter = (filterToToggle, checkedFilters, setCheckedFilters) => {
+    const enabledFilters = Array.from(checkedFilters);
+    if (enabledFilters.includes(filterToToggle)) {
+      const filterIndex = enabledFilters.indexOf(filterToToggle);
+      enabledFilters.splice(filterIndex, 1);
     } else {
-      enabledDocTypes.push(docTypeToToggle);
+      enabledFilters.push(filterToToggle);
     }
-    setCheckedFilters(enabledDocTypes);
+    setCheckedFilters(enabledFilters);
   };
 
   const submit = (e) => {
```

Comment:
When pressing the submit button the checked filters does not maintain for "Applicable to territory" when the page is refreshed (the tick disappears of what you have selected).  Doesn't look like this was intended at the moment looking at the code?

Reply:
I'm having some trouble replicating this, I'm currently seeing the below:

https://user-images.githubusercontent.com/9799940/139842195-8f0c55c9-a414-4b7d-814e-25e76a3039e0.mov 

Could you detail the steps you used to get the filters not persisting?

Reply:
I see the issue I think. So search for a product say "paracetamol" then do what you did. When you choose a product from the list using A -  Z etc. it works for me but not when searching for a product.

Reply:
👍 got it, I see what the issue is now, will put out a fix shortly.

Reply:
@gilbertsak Should be fixed in the latest commit